File: soluciones/E02-MonitorFrecuenciaCardiaca/Java/FrecuenciasCardiacasAleatorio/src/frecuenciascardiacasaleatorio/FrecuenciasCardiacasAleatorio.java
package frecuenciascardiacasaleatorio;

import java.util.Scanner;
import java.util.Random;
/**
 * @author Ricardo Espinosa y Juan Diego Guerrero Camargo
 */
public class FrecuenciasCardiacasAleatorio {
    public static void main(String[] args) {
        Scanner tc = new Scanner(System.in);
        String[] prefijos = {"Pedro", "Jesus", "Alfredo", "Claudio", "Juan", "Maria", "Luisa", "Mia", "Gabriela", "Omar", "Pablo", "Diego", "ismael"};
        Random random = new Random();
        String seguir = "s", seguir1 = "n";
        int anioActual = 2023, mesActual = 12, diaActual = 02, frecMax, dia = 0, cont = 0;
        double frecRango1, frecRango2;
        for (int i = 0; i < 1; i++) {
            String nombreAleatorio = NombreAleatorio(prefijos, random);
        }
        int diaNacimiento = (int) (Math.random() * 31) + 1;
        int mesNacimiento = (int) (Math.random() * 12) + 1;
        int anioNacimiento = 1900 + (int) (Math.random() * (2024 - 1900));
        int edad = (int) (Math.random() * 100) + 1;
        //Calculo de edad
        edad = anioActual - anioNacimiento;
        if (mesActual < mesNacimiento || (mesActual == mesNacimiento && diaActual < diaNacimiento)) {
            edad--;
        }
        //Frecuencia Maxima y Rango de frecuencia
        frecMax = 220 - edad;
        frecRango1 = frecMax * 0.50;
        frecRango2 = frecMax * 0.85;

        System.out.println("\nFrecuencias cardiacas de " + NombreAleatorio(prefijos, random));
        System.out.println("Fecha de Nacimiento: " + diaNacimiento + "/" + mesNacimiento + "/" + anioNacimiento);
        System.out.println("Edad: " + edad);
        System.out.println("Frecuencia cardiaca maxima: " + frecMax + " lpm");
        System.out.printf("%s %.0f %s %.0f %s", "Rango de frecuencia cardiaca: ", frecRango1, " lpm /", frecRango2, " lpm\n");
        //Arreglos
        int tamanoArreglo = 5;
        int[] historial = new int[tamanoArreglo];
        int[] hora = new int[tamanoArreglo];
        //Historial de frecuencias cardiacas
        System.out.println("\nDesea crear un historial de frecuencias cardiacas por un dia a su eleccion");
        System.out.println("s/n");
        seguir1 = tc.next();
        if (seguir1.equals("s")) {
             for (int i = 0; i < tamanoArreglo; i++) {
                historial[i] = 60 + random.nextInt(141);
                hora[i] = random.nextInt(25);
            }   
            System.out.println("\nDe que dia es el registro");
            System.out.println("[1] Lunes");
            System.out.println("[2] Martes");
            System.out.println("[3] Miercoles");
            System.out.println("[4] Jueves");
            System.out.println("[5] Viernes");
            System.out.println("[6] Sabado");
            System.out.println("[7] Domingo");
            dia = tc.nextInt();
            switch (dia) {
                case 1:
                    System.out.println("\nHistorial de frecuencias cardiacas de " + NombreAleatorio(prefijos, random) + " del dia Lunes");
                    break;
                case 2:
                    System.out.println("\nHistorial de frecuencias cardiacas de " + NombreAleatorio(prefijos, random) + " del dia Martes");
                    break;
                case 3:
                    System.out.println("\nHistorial de frecuencias cardiacas de " + NombreAleatorio(prefijos, random) + " del dia Miercoles");
                    break;
                case 4:
                    System.out.println("\nHistorial de frecuencias cardiacas de " + NombreAleatorio(prefijos, random) + " del dia Jueves");
                    break;
                case 5:
                    System.out.println("\nHistorial de frecuencias cardiacas de " + NombreAleatorio(prefijos, random) + " del dia Viernes");
                    break;
                case 6:
                    System.out.println("\nHistorial de frecuencias cardiacas de " + NombreAleatorio(prefijos, random) + " del dia Sabado");
                    break;
                case 7:
                    System.out.println("\nHistorial de frecuencias cardiacas de " + NombreAleatorio(prefijos, random) + " del dia Domingo");
                    break;
            }
            System.out.printf("%s \t %s \n", "Frecuencia ", " Hora");
            for (int i = 0; i < tamanoArreglo; i++) {
                System.out.printf("%d %s \t\t %d \n", historial[i], " lpm", hora[i]);
            }
        }
        System.out.println("\nInforme terminado");
        
    }
    private static String NombreAleatorio(String[] prefijos, Random random) {
        String nombreAleatorio = prefijos[random.nextInt(prefijos.length)];
        return nombreAleatorio;
    }

}

File: soluciones/E02-MonitorFrecuenciaCardiaca/Python/FrecuenciasCardiacasAleatorio.py
import random

def nombre_aleatorio(prefijos, random_generator):
    return random_generator.choice(prefijos)

# Contador de frecuencias cardiacas
prefijos = ["Pedro", "Jesus", "Alfredo", "Claudio", "Juan", "Maria", "Luisa", "Mia", "Gabriela", "Omar", "Pablo", "Diego", "ismael"]
random_generator = random.Random()

anio_actual, mes_actual, dia_actual = 2023, 12, 2
frec_max, dia, cont = 0, 0, 0

for i in range(1):  # Solo se ejecuta una vez, como en el bucle original de Java
    nombre_aleatorio = nombre_aleatorio(prefijos, random_generator)

dia_nacimiento = random_generator.randint(1, 31)
mes_nacimiento = random_generator.randint(1, 12)
anio_nacimiento = random_generator.randint(1900, 2023)
edad = anio_actual - anio_nacimiento
if mes_actual < mes_nacimiento or (mes_actual == mes_nacimiento and dia_actual < dia_nacimiento):
    edad -= 1

# Frecuencia Máxima y Rango de frecuencia
frec_max = 220 - edad
frec_rango1 = frec_max * 0.50
frec_rango2 = frec_max * 0.85

print("\nFrecuencias cardiacas de", nombre_aleatorio)
print("Fecha de Nacimiento:", f"{dia_nacimiento}/{mes_nacimiento}/{anio_nacimiento}")
print("Edad:", edad)
print("Frecuencia cardíaca máxima:", frec_max, "lpm")
print("Rango de frecuencia cardíaca:", f"{frec_rango1} lpm / {frec_rango2} lpm\n")

# Arreglos
tamano_arreglo = 5
historial = [random_generator.randint(60, 200) for _ in range(tamano_arreglo)]
hora = [random_generator.randint(0, 24) for _ in range(tamano_arreglo)]

# Historial de frecuencias cardiacas
seguir1 = input("¿Desea crear un historial de frecuencias cardiacas por un día a su elección? (s/n): ")
if seguir1.lower() == "s":
    for i in range(tamano_arreglo):
        historial[i] = random_generator.randint(60, 200)
        hora[i] = random_generator.randint(0, 24)

    print("\n¿De qué día es el registro?")
    print("[1] Lunes")
    print("[2] Martes")
    print("[3] Miércoles")
    print("[4] Jueves")
    print("[5] Viernes")
    print("[6] Sábado")
    print("[7] Domingo")
    dia = int(input())
    dias_semana = ["Lunes", "Martes", "Miércoles", "Jueves", "Viernes", "Sábado", "Domingo"]

    print(f"\nHistorial de frecuencias cardiacas de {nombre_aleatorio} del día {dias_semana[dia - 1]}")
    print("{:<15} {}".format("Frecuencia", "Hora"))
    for i in range(tamano_arreglo):
        print("{:<15} {}".format(str(historial[i]) + " lpm", hora[i]))

print("\nInforme terminado")
